sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -29,13 +30,57 @@ import {
   SuiObjectRef,
   getObjectReference,
   Coin,
+  SuiEventFilter,
+  SuiEventEnvelope,
+  SubscriptionId,
 } from '../types';
 import { SignatureScheme } from '../cryptography/publickey';
+import { Client as WsRpcClient} from 'rpc-websockets';
 
 const isNumber = (val: any): val is number => typeof val === 'number';
 const isAny = (_val: any): _val is any => true;
 
+const httpRegex = new RegExp('^http');
+const portRegex = new RegExp(':[0-9]{1,5}$');
+export const getWebsocketUrl = (httpUrl: string, port?: number): string => {
```

Comment:
Rather than just doing string manipulation, I'd create a `URL` object that you manipulate:

```ts
const url = new URL(httpUrl);
url.protocol = url.protocol.replace('http', 'ws');
url.port = port;
return url.toString();
```